src/braket/experimental/autoqasm/program/program.py

```diff
@@ -87,6 +88,65 @@ class ProgramMode(Enum):
     """For program conversion inside a context where only pulse operations are allowed."""
 
 
+class MainProgram(SerializableProgram):
```

Comment:
I'm not sure I like `MainProgram` as the class name here. Suggestions welcome.

Reply:
+1 I think it's good, agree it could be better. But let's not block on it (and I don't have a suggestion offhand). Let's add an action item to think about in a follow up PR?